runners/core-java/src/main/java/org/apache/beam/runners/core/SimpleDoFnRunner.java

```diff
@@ -589,8 +592,16 @@ public void outputWindowedValue(
             PaneInfo pane) {}
 
         @Override
-        public <T> T sideInput(PCollectionView<T> view, BoundedWindow mainInputWindow) {
-          return context.sideInput(view, mainInputWindow);
+        public <SideOutputT> void sideOutputWindowedValue(
+            TupleTag<SideOutputT> tag,
+            SideOutputT output,
+            Instant timestamp,
+            Collection<? extends BoundedWindow> windows,
+            PaneInfo pane) {}
```

Comment:
Seems like this (and the above which is duplicates) should both throw. A `DoFn` is not permitted to output to a different window. Silently dropping data seems a cruel way to enforce this.